### ISSUE TITLE
$import (<<<) bypasses harmony proxy traps

### ISSUE THREAD

**Issue by naturalethic** [gkz/LiveScript#791]
The implementation of $import, by using {}.hasOwnProperty.apply, bypasses the Proxy.hasOwn trap.  $import should be reimplemented in a way that does not do this.

**Comment by vendethiel** [comment]
...Which would include? :)

**Comment by naturalethic** [comment]
I'm sorta curious about the rationale for doing it the current way.  The fix would be to just do the obvious:
```livescript
  function import$(obj, src){
    for (var key in src) if (src.hasOwnProperty(key)) obj[key] = src[key];
    return obj;
  }
```

**Comment by gkz** [comment]
Maybe in case "hasOwnProperty" is define as an object property?

```
> o = {a:1}
Object {a: 1}
> o.hasOwnProperty("a")
true
> o.hasOwnProperty("b")
false
> o.hasOwnProperty = () => true
() => true
> o.hasOwnProperty("b")
true
> own = {}.hasOwnProperty
hasOwnProperty() { [native code] }
> own.call(o, "b")
false
```

**Comment by naturalethic** [comment]
My assumption about this is wrong.  I jumped the gun on reporting it.  A pure test case is working as expected, and the only traps called are `enumerate` and `ownKeys`.  I'm digging deeper...

**Comment by naturalethic** [comment]
I've solved this.  I needed to implement another trap.  In my 'pure' test, the call was going to the target silently, but in my specific application case, I actually reference properties on an object that is not the proxy target.  Thus the uncaught trap was trying to access on the target rather than the second object.

The method trap I needed to trap was: `getOwnPropertyDescriptor`

```livescript
    get-own-property-descriptor: (target, key) ->
      Object.get-own-property-descriptor non-target-object, key
```

Closing the issue.
